sample-std/src/lib.rs

```diff
@@ -0,0 +1,542 @@
+pub use quickcheck::{Arbitrary, Gen, TestResult, Testable};
+use std::{marker::PhantomData, ops::Range};
+
+pub use rand;
+use rand::{
+    distributions::uniform::SampleUniform, prelude::Distribution, seq::SliceRandom, Rng,
+    SeedableRng,
+};
+
+pub mod recursive;
+
+/// [Random] represents a PRNG.
+///
+/// It is a reimplementation of [quickcheck::Gen], which does not export the
+/// methods we need to properly generate random values.
+///
+/// It is unspecified whether this is a secure RNG or not. Therefore, callers
+/// should assume it is insecure.
+pub struct Random {
+    pub rng: rand::rngs::SmallRng,
```

Comment:
Honestly, because this code is derived from `Gen` in `quickcheck`. I'd like to get to the point where a single seed can be used to recover the initial sampled value, but that will require a rework of the macro that is still in progress (it's currently based off of `Arbitrary`, which cannot support this due to its design).

Reply:
This is a good point though; I'll look into the various other RNGs and see if there's one that better fits the eventual goals here.